src/test/kotlin/uk/gov/dluhc/printapi/messaging/ApplicationRemovedMessageListenerTest.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.dluhc.printapi.messaging
+
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.kotlin.await
+import org.junit.jupiter.api.Test
+import uk.gov.dluhc.printapi.config.IntegrationTest
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildCertificate
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildDelivery
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildPrintRequest
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildApplicationRemovedMessage
+import java.time.LocalDate
+import java.util.concurrent.TimeUnit
+
+internal class ApplicationRemovedMessageListenerTest : IntegrationTest() {
+
+    @Test
+    fun `should process message and set delivery info removal date`() {
+        // Given
+        val certificate = buildCertificate(
+            issueDate = LocalDate.of(2023, 4, 1), // to include 3 bank holidays in calculation
+            printRequests = listOf(
+                buildPrintRequest(delivery = buildDelivery()),
+                buildPrintRequest(delivery = buildDelivery())
+            )
+        )
+        certificateRepository.save(certificate)
+        val payload = buildApplicationRemovedMessage(
+            sourceReference = certificate.sourceReference!!,
+            gssCode = certificate.gssCode!!
+        )
+        val expectedInitialRemovalDate = LocalDate.of(2023, 5, 16)
```

Comment:
Minor, and I know you kinda mentioned it above, but might be worth a comment on the expectedInitialRemovalDate explaining why its 16th May 2023
```
// Certificate issue date is 01/04/2023. 28 working days from that date is 16/05/23 because there are xx weekend days, plus xx bank holidays (Good Friday and Easter Monday)
```
Something like that?

Reply:
Sorry but I don't think that's necessary (given I alluded to it above and I've added quite a bit of kdoc in places).

Reply:
We might have to agree to disagree on that 😁 👍 
Yes you've alluded to it above, yes you have plenty of really good kdoc; but from reading the test alone I'm not sure it's as obvious as it could be ¯\_(ツ)_/¯
But I'm happy to stand down on this 😁